fabric-events-lifecycle-v0/src/main/java/net/fabricmc/fabric/api/event/server/ServerSaveCallback.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.server;
+
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.world.level.LevelProperties;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * A callback for the server's saving.
+ */
+public interface ServerSaveCallback {
+	/**
+	 * An event invoked when the server's vanilla saving logic has completed.
```

Comment:
What "saving" means in this context should also be clarified

```suggestion
	 * An event invoked when the vanilla server's saving logic has completed.
```

Reply:
No. The server can have modded logic and vanilla logic. We aren't targeting vanilla server but vanilla logic here. Saving completion is that all components (chunks, level properties, etc) have been saved.